impeller/renderer/backend/vulkan/fence_waiter_vk.cc

```diff
@@ -101,76 +95,94 @@ void FenceWaiterVK::Main() {
     // critical section. Copy the array of entries and immediately unlock the
     // mutex.
     WaitSet wait_set = wait_set_;
-
-    const auto terminate = terminate_;
-
     lock.unlock();
 
+    const auto terminate = terminate_;
```

Comment:
Also do the read of `terminate_` while holding `wait_set_mutex_` here.

It may be easier to reason about this code if a scope is used to control where the `wait_set_mutex_` is held instead of calling `lock.unlock()`.

Reply:
Sounds reasonable. Done.